app/forms/idv/finance_form.rb

```diff
@@ -39,6 +39,13 @@ def finance_type_valid
       end
     end
 
+    def ccn_valid
+      return true unless finance_type == :ccn
+      unless finance_account =~ /^\d{8}$/
+        errors.add :finance_account, I18n.t('idv.errors.invalid_ccn')
+      end
+    end
```

Comment:
That does not quite work, because there doesn't seem to be a way to create a custom error message without also creating a custom validator class. The CCN error is so specific that having a high quality message (rather than simply "is invalid") seems ideal.

I have, however, refactored to put the `if` logic into the `validate` call itself.